include/seqan/blast/blast_report_out.h

```diff
@@ -0,0 +1,893 @@
+ *
+ * @include demos/blast/blast_out_example.report
+ *
+ * @see BlastRecord
+ */
+
+template <typename TBlastIOContext = BlastIOContext<> >
+using BlastReportOut = FormattedFile<BlastReport, Output, TBlastIOContext>;
+
+// ----------------------------------------------------------------------------
+// Class MagicHeader
+// ----------------------------------------------------------------------------
+
+template <typename T>
+struct MagicHeader<BlastReport, T> :
+    public MagicHeader<Nothing, T> {};
+
+// ----------------------------------------------------------------------------
+// Class FileExtensions
+// ----------------------------------------------------------------------------
+
+template <typename T>
+struct FileExtensions<BlastReport, T>
+{
+    static constexpr char const * VALUE[5] =
+    {
+        ".blast",
+        ".m0",
+        ".bm0"
+    };
+};
+
+template <typename T>
+constexpr char const * FileExtensions<BlastReport, T>::VALUE[5];
+
+// ----------------------------------------------------------------------------
+// Metafunction FormattedFileContext
+// ----------------------------------------------------------------------------
+
+template <typename TContext, typename TStorageSpec>
+struct FormattedFileContext<FormattedFile<BlastReport, Output, TContext>, TStorageSpec>
+{
+    typedef TContext Type;
+};
+
+// ----------------------------------------------------------------------------
+// Metafunction FileFormats
+// ----------------------------------------------------------------------------
+
+template <typename TSpec>
+struct FileFormat<FormattedFile<BlastReport, Output, TSpec> >
+{
+    typedef BlastReport Type;
+};
+
+// ============================================================================
+// Metafunctions
+// ============================================================================
+
+// ============================================================================
+// Functions
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// Function guessFormat()
+// ----------------------------------------------------------------------------
+
+template <typename TSpec>
+inline bool guessFormat(FormattedFile<BlastReport, Output, TSpec> &)
+{
+    return true;
+}
+
+// ----------------------------------------------------------------------------
+// Function _numberOfDigits
+// ----------------------------------------------------------------------------
+
+template <typename T>
+constexpr typename std::enable_if<std::is_integral<T>::value, T>::type
+_numberOfDigits(T const number)
+{
+    return (number == 0) ? 1 : static_cast<T>(std::floor(std::log10(number))+1);
+}
+
+// ----------------------------------------------------------------------------
+// some string constants
+// ----------------------------------------------------------------------------
+
+constexpr const char *
+_blastReference()
```

Comment:
It is the same really, the only difference is that to prevent global inclusion you would have to make the static const cstring a member of a templated struct (which is a hack by itself).